Adopt ASCIILiterals in makeString, tryMakeString, operator+ calls inside WTF

zdobersek opened the pull request:
#### 98c664538460ba8b172b12892675c7a00a728de9
<pre>
Adopt ASCIILiterals in makeString, tryMakeString, operator+ calls inside WTF
<a href="https://bugs.webkit.org/show_bug.cgi?id=251981">https://bugs.webkit.org/show_bug.cgi?id=251981</a>

Reviewed by Darin Adler.

Convert different cases of makeString(), tryMakeString() and StringAppend-producing
operator+ calls from using native C string literals to using more-desired ASCIILiterals.

* Source/WTF/wtf/Logger.cpp:
(WTF::Logger::LogSiteIdentifier::toString const):
* Source/WTF/wtf/Logger.h:
(WTF::Logger::logVerbose):
* Source/WTF/wtf/MediaTime.cpp:
(WTF::MediaTime::toString const):
* Source/WTF/wtf/URL.cpp:
(WTF::URL::setHost):
(WTF::URL::setHostAndPort):
(WTF::URL::setUser):
(WTF::URL::setPassword):
(WTF::URL::setQuery):
(WTF::URL::stringCenterEllipsizedToLength const):
(WTF::URL::fileURLWithFileSystemPath):
* Source/WTF/wtf/URLParser.cpp:
(WTF::URLParser::addNonSpecialDotSlash):

Canonical link: <a href="https://commits.webkit.org/260153@main">https://commits.webkit.org/260153@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0a3480308180f6e7087bdd953b147406c64717de

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107312 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16369 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40127 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116468 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/115912 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/17795 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/7607 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99468 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113076 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13428 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96529 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/41044 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95332 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/28164 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/82813 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/96685 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9400 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29508 "Passed tests") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/38/builds/96114 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/7394 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10044 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/6483 "Passed tests") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/38/builds/96114 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15553 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/49088 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/104949 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7023 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/11598 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/26008 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

fujii commented on Source/WTF/wtf/RefCounted.cpp:
```diff
@@ -21,8 +21,19 @@
 bool RefCountedBase::areThreadingChecksEnabledGlobally { false };
 
+#if 0
+static bool test()
```
Remove this test code.

darinadler commented on Source/WTF/wtf/MediaTime.cpp:
```diff
@@ -550,10 +550,10 @@ void MediaTime::dump(PrintStream& out) const
 String MediaTime::toString() const
 {
-    const char* invalid = isInvalid() ? ", invalid" : "";
+    ASCIILiteral invalid = isInvalid() ? ", invalid"_s : ""_s;
```
I like auto in cases like this.

zdobersek replied:
Applied.

zdobersek committed:
Adopt ASCIILiterals in makeString, tryMakeString, operator+ calls inside WTF
https://bugs.webkit.org/show_bug.cgi?id=251981

Reviewed by Darin Adler.

Convert different cases of makeString(), tryMakeString() and StringAppend-producing
operator+ calls from using native C string literals to using more-desired ASCIILiterals.

* Source/WTF/wtf/Logger.cpp:
(WTF::Logger::LogSiteIdentifier::toString const):
* Source/WTF/wtf/Logger.h:
(WTF::Logger::logVerbose):
* Source/WTF/wtf/MediaTime.cpp:
(WTF::MediaTime::toString const):
* Source/WTF/wtf/URL.cpp:
(WTF::URL::setHost):
(WTF::URL::setHostAndPort):
(WTF::URL::setUser):
(WTF::URL::setPassword):
(WTF::URL::setQuery):
(WTF::URL::stringCenterEllipsizedToLength const):
(WTF::URL::fileURLWithFileSystemPath):
* Source/WTF/wtf/URLParser.cpp:
(WTF::URLParser::addNonSpecialDotSlash):

Canonical link: https://commits.webkit.org/260153@main
